### ISSUE TITLE
Training procedure?

### ISSUE THREAD

**Issue by waleedrazakhan92** [nyukat/breast_cancer_classifier#33]
Hi! I would like to train your model on a dummy dataset. Can you guide me or better yet provide some helper code.? Specifically for the view wise model for a single image. The model outputs a csv file with the resulting probabilities for malignant/benign, so i'm assuming that the training dataset would be images with binary labels for malignant and benign classes. Am I assuming right?

**Comment by waleedrazakhan92** [comment]
Hi,

Thank you for your response. I also noticed in the sample .ipynb you provided, when you load the model you specifically set the view parameter as 'L-CC', which implies that you're loading 'L-CC' model. And also during inference you give an 'L-CC'  view image, which obviously makes sense. So if i want to run inference on 'R-CC' or 'L-MLO' view image, do i have to load that view-specific model for each specific image?  

Another question is that when i see the summary of that model (L-CC) model, it mentions output_features=4, and also when the inference is run on the model it outputs 4 values which you manipulate to get two output probabilities for malignant and benign. I didn't understand the very line ('np.exp(y_hat.cpu().detach().numpy())[:, :2, 1]') which converts those 4 tensor values of 'yhat' into the malignant/benign probabilities. 

Also, lets say i want to train this model on my own dataset, do i have to train 4 models for each view separately?  and then somehow combine the results of those 4 models? and also would each model would train on only view specific images (like the L-CC model would only run on L-CC images and R-CC would train on R-CC images and so on)?

**Comment by waleedrazakhan92** [comment]
Hi,

Thank you for your response. I also noticed in the sample .ipynb you provided, when you load the model you specifically set the view parameter as 'L-CC', which implies that you're loading 'L-CC' model. And also during inference you give an 'L-CC'  view image, which obviously makes sense. So if i want to run inference on 'R-CC' or 'L-MLO' view image, do i have to load that view-specific model for each specific image?  

Another question is that when i see the summary of that model (L-CC) model, it mentions output_features=4, and also when the inference is run on the model it outputs 4 values which you manipulate to get two output probabilities for malignant and benign. I didn't understand the very line ('np.exp(y_hat.cpu().detach().numpy())[:, :2, 1]') which converts those 4 tensor values of 'yhat' into the malignant/benign probabilities. 

Also, lets say i want to train this model on my own dataset, do i have to train 4 models for each view separately?  and then somehow combine the results of those 4 models? and also would each model would train on only view specific images (like the L-CC model would only run on L-CC images and R-CC would train on R-CC images and so on)?

**Comment by waleedrazakhan92** [comment]
Hi,

Thank you for your response. I also noticed in the sample .ipynb you provided, when you load the model you specifically set the view parameter as 'L-CC', which implies that you're loading 'L-CC' model. And also during inference you give an 'L-CC' view image, which obviously makes sense. So if i want to run inference on 'R-CC' or 'L-MLO' view image, do i have to load that view-specific model for each specific image?

Another question is that when i see the summary of that model (L-CC) model, it mentions output_features=4, and also when the inference is run on the model it outputs 4 values which you manipulate to get two output probabilities for malignant and benign. I didn't understand the very line ('np.exp(y_hat.cpu().detach().numpy())[:, :2, 1]') which converts those 4 tensor values of 'yhat' into the malignant/benign probabilities.

Also, lets say i want to train this model on my own dataset, do i have to train 4 models for each view separately? and then somehow combine the results of those 4 models? and also would each model would train on only view specific images (like the L-CC model would only run on L-CC images and R-CC would train on R-CC images and so on)?

**Comment by kjgeras** [comment]
> Thank you for your response. I also noticed in the sample .ipynb you provided, when you load the model you specifically set the view parameter as 'L-CC', which implies that you're loading 'L-CC' model. And also during inference you give an 'L-CC' view image, which obviously makes sense. So if i want to run inference on 'R-CC' or 'L-MLO' view image, do i have to load that view-specific model for each specific image?
Yes, as far as I remember.

> Another question is that when i see the summary of that model (L-CC) model, it mentions output_features=4, and also when the inference is run on the model it outputs 4 values which you manipulate to get two output probabilities for malignant and benign. I didn't understand the very line ('np.exp(y_hat.cpu().detach().numpy())[:, :2, 1]') which converts those 4 tensor values of 'yhat' into the malignant/benign probabilities.
I don't understand what exactly you are asking. Please follow what the model does in the example we provided.

> Also, lets say i want to train this model on my own dataset, do i have to train 4 models for each view separately? and then somehow combine the results of those 4 models? and also would each model would train on only view specific images (like the L-CC model would only run on L-CC images and R-CC would train on R-CC images and so on)?
Yes, that is what you should do if you wanted to train that model. I think this model https://github.com/nyukat/GMIC is actually probably better for what you are trying to do.
